### DISCUSSION TITLE
Should plugins collect files from `files` automatically?

### DISCUSSION THREAD

**Discussion by pmoravec** [sosreport/sos#3647] [Ideas]
`files` stands for a plugin's enabledness trigger. Which means each such file is very typically present (only) on a system with given component/product/technology the plugin is relevant to.

Should the plugin *automatically* collect those files? Similarly like we do for `services` collecting its status and journal logs (or for kmods collecting `modinfo`)? See [add_default_collections](https://github.com/sosreport/sos/blob/main/sos/report/plugins/__init__.py#L3275).

This derived from discussion in https://github.com/sosreport/sos/pull/3645 where we worked on that assumption. My original attitude was "we should collect it by default". But now I lean towards current approach, not collect by default. The main reason is the file is a symptom of installation of some component/tech/.., which surely does not mean it contains interesting debugging info. E.g. some triggers are `/dev` or `/sys/block` or `/var/spool/abrt` or `/var/www` we should avoid automatic collection.

We can have `add_forbidden_path` in such cases but I see it risky - we should not automatically collect some raw data the user has not specified.

Let me know if you have a different opinion and would like to automatically collect copyspecs from plugins' `files`.

**Comment by jcastill** [comment]
I'm with you here, automatic collection could be a problem because we'll end up having to check if the entry is a directory, a binary, a cert, etc... rather than a log or a config file.

**Comment by arif-ali** [comment]
God points @pmoravec, if we have triggers as `/dev/` we don't want to be collecting all of that, and doing exclusions would be cumbersome and very problematic.

There are 2 ways we can accommodate this imo,

1. if the `files` dict doesn't have anything cumbersome such as the example in #3581, where I discussed this internally, and we could just call `super().setup()` which can then collect the stuff from `files`.
2. Or, like we do today, add the files needed in `add_copy_spec`, which then avoids the extra collections when not needed.

Always good to think it out :)

**Comment by dnegreira** [comment]
I agree with you @pmoravec, we should not be collecting those files by default as they are used only to verify that they are there for the plugin to run.

**Comment by TurboTurtle** [comment]
> I agree with you @pmoravec, we should not be collecting those files by default as they are used only to verify that they are there for the plugin to run.

My thoughts as well. I don't think we gain enough by doing this to offset the potential for new problems.

**Comment by pmoravec** [comment]
As we reach a consensus here, I am closing the discussion.
